extensions/indexes/lucene/src/main/java/org/exist/indexing/lucene/LuceneFacetConfig.java

```diff
@@ -92,6 +98,19 @@ protected void processResult(Sequence result, Document luceneDoc) throws XPathEx
         }
     }
 
+    private void createHierarchicalFacet(Document luceneDoc, Sequence seq) throws XPathException {
+        final List<String> paths = new ArrayList<>(seq.getItemCount());
```

Comment:
As you eventually need an array and the size of the array is known, it would be much more efficient to just use an array rather instead of an array list.

Reply:
Previously an array was used, but this caused an exception being thrown by lucene if `getStringValue()` returned an empty string (which is a valid value in XQuery but rejected by lucene). We thus have to build the list dynamically.

Reply:
@adamretter Could you please mark this and your requested changes as resolved (assuming they are)?

Reply:
@wolfgangmm Sure... but why not:

```java
final String[] paths = new String[seq.getItemCount()];
int i = 0;
for (final SequenceIterator i = seq.unorderedIterator(); i.hasNext(); ) {
  final String value = i.nextItem().getStringValue();
  if (!value.isEmpty() > 0) {
    paths[i++] = value;
  }
}
if (i > 0) {
  luceneDoc.add(new FacetField(dimension, Arrays.copyOf(paths, i)));
}
```

Reply:
@joewiz it's not resolved yet ;-)

Reply:
@wolfgangmm Do you agree with @adamretter's suggestion? I've resolved the merge conflict, so this is the only remaining item before the PR can be merged.

Reply:
We cannot have empty items in the array. i'm not sure if the suggestion considers this,  but anyway, I absolutely don't see any benefit in this formulation. It's not that this code is critical in any way.

Reply:
Excellent, thanks. We’ll move ahead as is.